src/Infrastructure/Persistence/Doctrine/Migrations/Version20231031100605.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure\Persistence\Doctrine\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20231031100605 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Migrate dailyrange data';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"');
+        $this->addSql('
+            UPDATE period SET start_datetime = ro.start_date, end_datetime = ro.end_date
+            FROM measure AS m
+            INNER JOIN location AS l ON l.uuid = m.location_uuid
+            INNER JOIN regulation_order AS ro ON ro.uuid = l.regulation_order_uuid
+            WHERE ro.start_date <> NULL
+            AND ro.end_date <> NULL
+        ');
+        $this->addSql('
+            INSERT INTO dailyrange (uuid, applicable_days, period_uuid)
+            SELECT public.uuid_generate_v4(), p.applicable_days, p.uuid
+            FROM period AS p
+            WHERE p.applicable_days <> NULL
+        ');
+        $this->addSql('
+            INSERT INTO timeslot (uuid, start_time, end_time, daily_range_uuid)
+            SELECT public.uuid_generate_v4(), p.start_time, p.end_time, d.uuid
+            FROM dailyrange AS d
+            INNER JOIN period AS p ON p.uuid = d.period_uuid
+            WHERE p.start_time <> NULL
+            AND p.end_time <> NULL
+        ');
+
+        $this->addSql('ALTER TABLE period DROP applicable_days');
+        $this->addSql('ALTER TABLE period DROP start_time');
+        $this->addSql('ALTER TABLE period DROP end_time');
```

Comment:
Je me souviens de la bonne pratique qu'on appliquait dans la boîte où j'ai fait mon stage de fin d'études : 

On faisait 3 PR _distinctes _qu'on mergeait et déployait dans l'ordre, après avoir vérifié que chacune produisait l'effet désiré

1. Création des nouveaux champs en nullable
2. Remplissage des nouveaux champs, on vérifie leur contenu
3. Suppression des anciens champs et passage des nouveaux en not null (si pertinent)

Car sinon dans l'état de cette migration, s'il y a un cas pas envisagé, on perd d'office la donnée et on ne peut pas "rollback" en supprimant juste les nouveaux champs, puisque les anciens ont déjà été supprimés